lib/model/narrow.dart

```diff
@@ -19,6 +19,18 @@ sealed class Narrow {
 /// A non-interleaved narrow, completely specifying a place to send a message.
 sealed class SendableNarrow extends Narrow {
   MessageDestination get destination;
+
+  static SendableNarrow ofMessage(Message message, {required int selfUserId}) {
+    switch (message) {
+      case StreamMessage(:var streamId, :var subject):
+        return TopicNarrow(streamId, subject);
+      case DmMessage(:var displayRecipient):
+        return DmNarrow(
+          allRecipientIds: displayRecipient.map((r) => r.id).toList(),
+          selfUserId: selfUserId,
+        );
+    }
+  }
```

Comment:
I wasn't sure if this would be best as a static method on `SendableNarrow`, or a helper function defined at toplevel, or even a method on `Message`.

Reply:
Makes sense.